### PR TITLE
Update test_models.py

### PR DESCRIPTION
**Pull request by brianhou0208**
## Description

New decoder was added in #926, but I found that UPerNet & DeepLabV3Plus missed some tests.

- [test_forward_backward](https://github.com/qubvel-org/segmentation_models.pytorch/blob/277af2f051e96bfe69e516e3c6eaddd44bcab740/tests/test_models.py#L107)
- [test_aux_output](https://github.com/qubvel-org/segmentation_models.pytorch/blob/277af2f051e96bfe69e516e3c6eaddd44bcab740/tests/test_models.py#L117)

This PR fixes some bugs and ensures that each decoder has been tested.

### CODE REVIEW

**Review by qubvel — APPROVED** [review]
Thanks for adding!

File: tests/test_models.py [hunk 1]
```diff
@@ -29,12 +29,11 @@ def get_sample(model_class):
         smp.PSPNet,
         smp.UnetPlusPlus,
         smp.MAnet,
-        smp.UPerNet,
     ]:
         sample = torch.ones([1, 3, 64, 64])
-    elif model_class == smp.PAN:
+    elif model_class == smp.PAN or model_class == smp.UPerNet:
         sample = torch.ones([2, 3, 256, 256])
-    elif model_class == smp.DeepLabV3:
```
Comment (qubvel) [inline tests/test_models.py:37]: Just curious why UPerNet requires a bigger input size.

Comment (brianhou0208) [reply tests/test_models.py:37]: Hi @qubvel ,

Actually, UPerNet don't need to use a larger size for testing,
but PyTorch's nn.BatchNorm2d does not support inputs with shape (B, C, H, W) == (1, C, 1, 1).

## Why:
In UPerNet, the [PSPModule](https://github.com/qubvel-org/segmentation_models.pytorch/blob/8def40f3b35026fa5ebc4b14ddab2feeb0010111/segmentation_models_pytorch/decoders/upernet/decoder.py#L17-L30) uses AdaptiveAvgPool2d, which results in outputs of:
`[(B, C, 1, 1), (B, C, 2, 2), (B, C, 3, 3), (B, C, 6, 6)]`

When model.train() mode, the Batch Normalization (BN) layers are activated.
If the batch size is 1, it triggers the following error:
`raise ValueError(f"Expected more than 1 value per channel when training, got input size {size}")`

[PSPNet](https://github.com/qubvel-org/segmentation_models.pytorch/blob/8def40f3b35026fa5ebc4b14ddab2feeb0010111/segmentation_models_pytorch/decoders/pspnet/decoder.py#L11-L12) also encounters this issue, but in that case, it skips the BN layer for the 1x1 size directly.

## Solution:

- Increase the batch size to 2 for specific models:
```
elif model_class in [smp.PSPNet, smp.UPerNet]:
    sample = torch.ones([2, 3, 64, 64])
```

- Set the batch size to 2 uniformly for all models:
```
if model_class in [
    smp.Unet,
    smp.Linknet,
    smp.FPN,
    smp.PSPNet,
    smp.UnetPlusPlus,
    smp.MAnet,
    smp.UPerNet,
]:
    sample = torch.ones([2, 3, 64, 64])
```

Comment (qubvel) [reply tests/test_models.py:37]: Thanks for clarifying! I suppose 
```
elif model_class in [smp.PSPNet, smp.UPerNet]:
    sample = torch.ones([2, 3, 64, 64])
```
Is better with a small comment in the code

Comment (qubvel) [reply tests/test_models.py:37]: Looks good

**Comment by brianhou0208** [comment]
Hi @qubvel ,
I think it's ready to merge.

## Update
- Add a new test size to avoid the PSPModule pooling size of 1x1 in PSPNet/UPerNet from triggering a PyTorch nn.BatchNorm2d size error.

**Comment by qubvel** [comment]
Thanks a lot for contribution!
